Sources/TuistCore/Utils/XCFrameworkMetadataProvider.swift

```diff
@@ -0,0 +1,62 @@
+import Basic
+import Foundation
+import TuistSupport
+
+enum XCFrameworkMetadataProviderError: FatalError, Equatable {
+    
+    case missingRequiredFile(AbsolutePath)
+
+    // MARK: - FatalError
+
+    var description: String {
+        switch self {
+        case let .missingRequiredFile(path):
+            return "The .xcframework at path \(path.pathString) doesn't contain an Info.plist. It's possible that the .xcframework was not generated properly or that got corrupted. Please, double check with the author of the framework."
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .missingRequiredFile:
+            return .abort
+        }
+    }
+}
+
+public protocol XCFrameworkMetadataProviding {
+    
+    /// Returns the available libraries for the xcframework at the given path.
+    /// - Parameter frameworkPath: Path to the xcframework.
+    func libraries(frameworkPath: AbsolutePath) throws -> [XCFrameworkInfoPlist.Library]
+    
+    /// Given a framework path and libraries it returns the path to its binary.
+    /// - Parameter frameworkPath: Framework path.
+    /// - Parameter libraries: Framework available libraries
+    func binaryPath(frameworkPath: AbsolutePath, libraries: [XCFrameworkInfoPlist.Library]) -> AbsolutePath
+}
+
+public class XCFrameworkMetadataProvider: XCFrameworkMetadataProviding {
```

Comment:
What about adding and integration tests for this with a real .xcframework?

Reply:
Added